### ISSUE TITLE
"Server access certificate expired on ." error when signing with ID card

### ISSUE THREAD

**Issue by jaakristioja** [open-eid/gentoo#67]
Using app-crypt/qdigidoc4-4.3.0 I'm getting the following error message, when trying to sign an asice container using Estonian ID card: `Server access certificate expired on . To renew the certificate please contact IT support team of your company. Additional information is available [sales@sk.ee](mailto:sales@sk.ee) or phone (+372) 610 1885`.

![image](https://github.com/open-eid/gentoo/assets/2684083/8a7e322d-2de9-4009-97ca-4bd4214d8c11)

What is this about? I'm a private person with no company and have no IT support team to contact. Please help.

Also note that the message seems to be missing the expiration date.

**Comment by metsma** [comment]
Server access certificate expired.
One option is to use in settings dialog disable server access certificate.
Or upgrade to 4.4.0

**Comment by jaakristioja** [comment]
First, this Gentoo overlay does not yet support 4.4.0.

Second, does upgrading to 4.4.0 have the exact same effect as disabling server access certificate or does it do something else?

Third, when I navigate to Settings -> Signing Services, mark the "Ignore server access certificate (IP based access)" checkbox and click "Close" (as there is no separate "Apply changes" or "Save settings" button), it still gives me the same error. When I navigate back to settings, the checkbox is shown unmarked.

**Comment by metsma** [comment]
I think there is PR #65 and the Gentoo overlay is community maintained repository.
The server access certificate was removed in DigiDoc 4.4.0.
I think it is possible to disable in digidocpp.conf file this also
https://github.com/open-eid/libdigidocpp/blob/v3.14.12/etc/digidocpp.conf.cmake#L26

**Comment by jaakristioja** [comment]
This overlay only provides dev-libs/libdigidocpp-3.16.0 and this option seems to be gone in https://github.com/open-eid/libdigidocpp/blob/v3.16.0/etc/digidocpp.conf.cmake so I presumed that it would not work and did not test it.

Upgrading to qdigidoc4 to 4.4.0 using the ebuild from PR #65 seems to be a valid workaround thou. I suppose this issue can be closed after #65 is merged. Thanks! 😃

**Comment by arvojarve** [comment]
Yep , #65 should be merged  I done it manually in my repo and now I can sign documents. TNX to author.

Even more, dev-libs/libdigidocpp should be upgraded to 3.17.1, this goes without any problem, and app-crypt/qdigidoc4
to 4.5.1, there #include \<optional\> should be added to qdigidoc4-4.5.1/client/Application.cpp, at least for gcc13.3.1

**Comment by lkraav** [comment]
Ran into this today as well. It's unclear what's blocking merging #65 is it just about team bandwidth, or is there something fundamentally off with the PR?

**Comment by lkraav** [comment]
I haven't tested, but in theory this can be closed with #65 merged?
